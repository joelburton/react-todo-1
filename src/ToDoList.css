.to-do-list {
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  font-size: 20px;
  
}