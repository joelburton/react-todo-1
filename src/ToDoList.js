import React, { useState } from 'react';
import './ToDoList.css';
import ToDoItem from "./ToDoItem";
import ToDoForm from "./ToDoForm";
import { v4 as uuid } from "uuid";

function ToDoList() {

  // set up an empty array of ToDos with state
  const [toDoItems, setToDoItems] = useState([]);

  // Passthrough function to add a new item to toDoItems
  const addToDo = toDo => {
    let newToDo = { ...toDo, id: uuid() };
    setToDoItems(ToDoItems => [...toDoItems, newToDo]);
  };

  //Passthrough function for the ToDoItem to remove it from state

const removeToDo = (ToDoId) => {
  const toDoItemsCopy = toDoItems.filter((i) => i.id !== ToDoId);
  setToDoItems(toDoItemsCopy);
};

// Create the ToDoList ready to render
  const renderToDoItems = () => {
    return (
      <div className="to-do-list" data-testid="ToDoList-list-container">
        {toDoItems.map(item => <ToDoItem id={item.id}
          key={item.id}
          text={item.text}
          removeToDo={removeToDo} />)}
      </div>
    );
  };

  //Render an add item form and the current list of ToDos.
  return (
    <div className="ToDo-List" data-testid="ToDoList-form-container">
      <ToDoForm addToDo={addToDo} />
      {renderToDoItems()}
    </div>
  );
};

export default ToDoList;