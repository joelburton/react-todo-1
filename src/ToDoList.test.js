import React from "react";
import { render, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import ToDoList from "./ToDoList";


function addToDoHelper(getByLabelText,queryByText, toDo) {

  const textInput = getByLabelText("To do:");
  const submitBox = queryByText("Add a new to do!");
  fireEvent.change(textInput, { target: { value: toDo }});
  fireEvent.click(submitBox);
}

it("renders without crashing", function() {
  render(<ToDoList />);
});

it("matches snapshot", function() {
  const {asFragment} = render(<ToDoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new to do item", function() {
  const { getByLabelText, queryByText } = render(<ToDoList />);

  // no to do yet
  expect(queryByText("Delete task")).not.toBeInTheDocument();

  addToDoHelper(getByLabelText, queryByText, "To Do Test 1");

  // item exists!
  expect(queryByText("To Do Test 1")).toBeInTheDocument();
});

it("can delete a to do item", function() {
  const { getByLabelText, queryByText, getAllByText } = render(<ToDoList />);

  // no to do yet
  expect(queryByText("Delete task")).not.toBeInTheDocument();

  addToDoHelper(getByLabelText, queryByText, "To Do Test 1");
  addToDoHelper(getByLabelText, queryByText, "To Do Test 2");

  // item exists!
  expect(queryByText("To Do Test 1")).toBeInTheDocument();
  expect(queryByText("To Do Test 2")).toBeInTheDocument();

  const deleteButtons = getAllByText("Delete task");

  fireEvent.click(deleteButtons[0]);
  expect(queryByText("To Do Test 1")).not.toBeInTheDocument();
});