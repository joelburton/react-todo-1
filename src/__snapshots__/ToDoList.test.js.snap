// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDo-List"
    data-testid="ToDoList-form-container"
  >
    <form>
      <label
        for="text"
      >
        To do: 
      </label>
      <input
        id="text"
        name="text"
        value=""
      />
      <button>
        Add a new to do!
      </button>
    </form>
    <div
      class="to-do-list"
      data-testid="ToDoList-list-container"
    />
  </div>
</DocumentFragment>
`;
